src/commands/createFunctionApp/stacks/FunctionAppStackStep.ts

```diff
@@ -20,6 +22,24 @@ export class FunctionAppStackStep extends AzureWizardPromptStep<IFunctionAppWiza
             if (!result) {
                 context.version = await promptForFuncVersion(context);
             } else {
+                const endofLifeDate = result.minorVersion.stackSettings.linuxRuntimeSettings?.endOfLifeDate;
+                const sixMonthsFromNow = new Date();
+                sixMonthsFromNow.setMonth(sixMonthsFromNow.getMonth() + 6);
+                const settingKey: string = 'endOfLifeWarning';
+                if (getWorkspaceSetting<boolean>('endOfLifeWarning')) {
+                    if (endofLifeDate) {
+                        if (compareDates(endofLifeDate, sixMonthsFromNow)) {
+                            const message = localize('endOfLife', "The chosen runtime stack has an end of support deadline coming up. After the deadline, function apps can be created and deployed, and existing apps continue to run. However, your apps won't be eligible for new features, security patches, performance optimizations, and support until you upgrade them");
+                            const result: MessageItem = await context.ui.showWarningMessage(message, { modal: true }, DialogResponses.learnMore, DialogResponses.dontWarnAgain);
+                            if (result === DialogResponses.learnMore) {
+                                await openUrl('https://learn.microsoft.com/en-us/azure/azure-functions/functions-versions?tabs=azure-cli%2Cwindows%2Cin-process%2Cv4&pivots=programming-language-csharp');
```

Comment:
Should be aka.ms link (extracted as constant and shared where it's used elsewhere). Aka.ms is our link shortening service. We do this so in the event we have to change a link later, we can do so without having to make code changes and do a release.